resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/hadooputils/HadoopBootstrapUtilSuite.scala

```diff
@@ -0,0 +1,227 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.features.hadooputils
+
+import java.io.File
+import java.util.UUID
+
+import scala.collection.JavaConverters._
+
+import com.google.common.io.Files
+import io.fabric8.kubernetes.api.model._
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.deploy.k8s.Constants._
+import org.apache.spark.deploy.k8s.SparkPod
+import org.apache.spark.deploy.k8s.features.KubernetesFeaturesTestUtils._
+import org.apache.spark.util.Utils
+
+class HadoopBootstrapUtilSuite extends SparkFunSuite with BeforeAndAfter{
+  private val sparkPod = SparkPod.initialPod()
+  private val hadoopBootstrapUtil = new HadoopBootstrapUtil
+  private var tmpDir: File = _
+  private var tmpFile: File = _
+
+  before {
+    tmpDir = Utils.createTempDir()
+    tmpFile = createTempFile(tmpDir, "contents")
+  }
+
+  after {
+    tmpFile.delete()
+    tmpDir.delete()
+  }
+
+  test("Testing bootstrapKerberosPod with file location of krb5") {
+    val dtSecretName = "EXAMPLE_SECRET_NAME"
+    val dtSecretItemKey = "EXAMPLE_ITEM_KEY"
+    val userName = "SPARK_USER_NAME"
+    val fileLocation = Some(tmpFile.getAbsolutePath)
+    val stringPath = tmpFile.toPath.getFileName.toString
+    val newKrb5ConfName = Some("/etc/krb5.conf")
+    val resultingPod = hadoopBootstrapUtil.bootstrapKerberosPod(
+      dtSecretName,
+      dtSecretItemKey,
+      userName,
+      fileLocation,
+      newKrb5ConfName,
+      None,
+      sparkPod)
+    val expectedVolumes = Seq(
+      new VolumeBuilder()
+        .withName(KRB_FILE_VOLUME)
+        .withNewConfigMap()
+          .withName(newKrb5ConfName.get)
+          .withItems(new KeyToPathBuilder()
+            .withKey(stringPath)
+            .withPath(stringPath)
+            .build())
+        .endConfigMap()
+        .build(),
+      new VolumeBuilder()
+        .withName(SPARK_APP_HADOOP_SECRET_VOLUME_NAME)
+        .withNewSecret()
+          .withSecretName(dtSecretName)
+          .endSecret()
+        .build()
+    )
+    assert(podHasVolumes(resultingPod.pod, expectedVolumes))
```

Comment:
If this assert (and others like it) fails, you'll get a pretty generic message ("false was not true" or some such).

Might be better to make `podHasVolumes` (and friends) assert with a more helpful message that shows at least the computed set.